### ISSUE TITLE
After update to vscode 1.82 the extension does not start: `Failed to parse URL`

### ISSUE THREAD

**Issue by zyrikby** [znck/grammarly#384]
Today, I've updated my vscode to the latest version (1.82), and the extension stopped working. Here is the output that I get:

```output
text checking session for "file:///home/yury/notes.md" is ready
node:internal/deps/undici/undici:11413
    Error.captureStackTrace(err, this);
          ^

TypeError: Failed to parse URL from /home/yury/.vscode/extensions/znck.grammarly-0.23.15/dist/server/tree-sitter.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11) {
  [cause]: TypeError [ERR_INVALID_URL]: Invalid URL
      at new NodeError (node:internal/errors:399:5)
      at URL.onParseError (node:internal/url:566:9)
      at new URL (node:internal/url:646:5)
      at new Request (node:internal/deps/undici/undici:7002:25)
      at fetch2 (node:internal/deps/undici/undici:10554:25)
      at Object.fetch (node:internal/deps/undici/undici:11411:18)
      at fetch (node:internal/process/pre_execution:248:25)
      at /home/yury/.vscode/extensions/znck.grammarly-0.23.15/dist/server/index.node.js:34:77331
      at /home/yury/.vscode/extensions/znck.grammarly-0.23.15/dist/server/index.node.js:34:77551
      at new Promise (<anonymous>) {
    input: '/home/yury/.vscode/extensions/znck.grammarly-0.23.15/dist/server/tree-sitter.wasm',
    code: 'ERR_INVALID_URL'
  }
}

Node.js v18.15.0
[Error - 10:54:29] Connection to server got closed. Server will not be restarted.
```

**Comment by BoDonkey** [comment]
Same. Tried reverting to 0.18.1 without luck.

**Comment by codetroll** [comment]
Same here, tried quite a few versions back to 0.18 - none of them works.

**Comment by yogidevendra** [comment]
I tried reverting vscode version. 
---
Version: 1.77.3
Commit: 704ed70d4fd1c6bd6342c436f1ede30d1cff4710
Date: 2023-04-12T09:41:48.638Z (4 mos ago)
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 22.4.0
Sandboxed: No
---

This plugin is working fine with the above mentioned vscode version.

**Comment by lestgabo** [comment]
confirmed. downgrading VS code to 1.77.3 worked for me

**Comment by BoDonkey** [comment]
For all of those reporting this issue, are you using the default-theme sidebar, or have you added a custom theme sidebar?

**Comment by vedpd11** [comment]
I am using default theme bar and getting the error. Even logout option is disappeared.
Attached screenshot
<img width="626" alt="image" src="https://github.com/znck/grammarly/assets/104493972/02034d56-4b2a-4c7b-8c8f-23c4d7fd5d3e">

**Comment by zyrikby** [comment]
After today's update to 1.82.1, the error still prevents the plugin to start:

```
Version: 1.82.1
Commit: 6509174151d557a81c9d0b5f8a5a1e9274db5585
Date: 2023-09-08T08:41:36.199Z
Electron: 25.8.0
ElectronBuildId: 23503258
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 5.15.0-79-generic
```

**Comment by vedpd11** [comment]
Pls see screenshot attached as well

<img width="1373" alt="image" src="https://github.com/znck/grammarly/assets/104493972/b9f91eb2-efe5-497a-ac76-4200c51da742">

<img width="1389" alt="image" src="https://github.com/znck/grammarly/assets/104493972/f9d16ccb-c226-4236-8376-a79dc1b3d1d9">

**Comment by jsulopzs** [comment]
If you download vscode with [this version](https://code.visualstudio.com/updates/v1_77), you can make grammarly work.

**Comment by RasmusLeupold** [comment]
This issue seems to be caused by the upgrade of Node.js from the major version of 16 to 18.

**Comment by vedpd11** [comment]
any solution for this other than downgrading ?

**Comment by codetroll** [comment]
As it seem that Grammaryly will drop support for the SDK january year, this plugin may not have a very long life anyway. [Grammarly will discontinue the Text Editor SDK](https://developer.grammarly.com/sdk-deprecation)

I switched to this one instead :  LTexX - LanguageTool grammar/spell checking](https://marketplace.visualstudio.com/items?itemName=valentjn.vscode-ltex)

**Comment by llimon** [comment]
As a paying customer of Grammarly, I hope VSCode integration will be fixed soon.

**Comment by adriancaruana** [comment]
> As it seem that Grammaryly will drop support for the SDK january year, this plugin may not have a very long life anyway. [Grammarly will discontinue the Text Editor SDK](https://developer.grammarly.com/sdk-deprecation)
> 
> I switched to this one instead : LTexX - LanguageTool grammar/spell checking](https://marketplace.visualstudio.com/items?itemName=valentjn.vscode-ltex)

Oh, wow, I had no idea about this! Ok, well no point hanging around here then. I'll give the LTexX extension a go, thanks!

**Comment by codetroll** [comment]
Yeah it is kinda sad to see the SDK go, I really liked the VS Code extension but am glad to have found the LTeX one instead.

**Comment by luongvo** [comment]
+1 looking for the fix soon

**Comment by joezhouchenye** [comment]
Sadly, LTeX isn't as good as Grammarly. It misses some spelling mistakes and it doesn't know some words.

**Comment by codetroll** [comment]
Perhaps not - but compared to have nothing it all, I'd say it is still ok. And again, if the SDK disappears we won't be able to have any Grammarly support anyway :(

**Comment by thomassuedbroecker** [comment]
> > As it seem that Grammaryly will drop support for the SDK january year, this plugin may not have a very long life anyway. [Grammarly will discontinue the Text Editor SDK](https://developer.grammarly.com/sdk-deprecation)
> > I switched to this one instead : LTexX - LanguageTool grammar/spell checking](https://marketplace.visualstudio.com/items?itemName=valentjn.vscode-ltex)
> 
> Oh, wow, I had no idea about this! Ok, well no point hanging around here then. I'll give the LTeX extension a go, thanks!
> 
> Edit: 30 mins in, and LTeX is a more than sufficient alternative to Grammarly. It's configurable and allows the selection of local dialects. Give it a go!

... true ... https://developer.grammarly.com/sdk-deprecation

"**Grammarly will discontinue the Text Editor SDK**
On Wednesday, January 10, 2024, the Grammarly Text Editor SDK will be discontinued. After that date, the SDK will no longer be able to offer Grammarly’s writing suggestions in your app.

First, thank you to the thousands of developers who used the Text Editor SDK and gave honest, constructive feedback as we launched Grammarly’s first developer product.

We made this decision based on learnings and input from the developer community, as well as to prioritize developing our core product offering in ways that go beyond writing assistance to ease many aspects of an organization’s communication workflow. For more information on Grammarly’s upcoming initiatives, watch our recent keynote: [Empowering the AI-Connected Workplace](https://www.youtube.com/watch?v=s42h4VHpqlo).

What this means for you
Grammarly for Developers, along with the SDK functionality and user experience, will be discontinued on January 10, 2024. We advise communicating this change to your users and removing the SDK from your application before that date.

If you currently have an active Grammarly for Developers application, the Text Editor SDK will continue to operate as normal until January 10, 2024."

We will not be accepting new registrations to Grammarly for Developers starting today. For those who recently registered and are considering a Text Editor SDK integration, you will have until July 26, 2023, to create a new app. Please note that the Text Editor SDK will no longer function in your app after January 10, 2024.

Questions? [Send us a note](https://support.grammarly.com/hc/en-us/requests/new#/resolve-a-technical-issue/text-editor-sdk-api-/)

**Comment by phyllisstein** [comment]
Another alternative is Grammarly's recently released [desktop apps](https://www.grammarly.com/desktop).

On my Mac, the app doesn't work _great_ with VSCode. It only checks the text I highlight, but each correction blurs focus and deselects everything. It's tolerable if you make peace with hitting <kbd>⌘+A</kbd> over and over.

Give it a try long enough to send them feedback with a bug report. They were very responsive when I complained about other Electron apps, writing back and eventually fixing the bug.

**Comment by znck** [comment]
Fixed today in v0.24.0
